.github/workflows/unit_tests.yml

```diff
@@ -45,19 +51,20 @@ jobs:
       - name: Run unit tests
         run: ./gradlew test coverage
 
-      - name: Save PR number to file
-        run: echo ${{ github.event.number }} > PR_NUMBER.txt
-
-      - name: Archive PR number
-        uses: actions/upload-artifact@v4
-        with:
-          name: pr_number
-          path: PR_NUMBER.txt
-
-      - name: Archive reports
-        uses: actions/upload-artifact@v4
```

Comment:
Does this still work without having to upload/download the artifact? Does the sonar find them automatically?